test_fakeredis.py

```diff
@@ -3078,6 +3078,36 @@ def _listen(pubsub, q):
         received = q.get()
         self.assertEqual(received['data'], msg)
 
+    @attr('slow')
+    def test_pubsub_run_in_thread(self):
+        q = Queue()
+
+        def _queue_msg(msg, q):
+            q.put(msg)
+
+        pubsub = self.redis.pubsub()
+        pubsub.subscribe(**{"channel": lambda e: _queue_msg(e, q)})
+        pubsub_thread = pubsub.run_in_thread()
+
+        msg = b"Hello World"
+        self.redis.publish("channel", msg)
+
+        sleep(1)
```

Comment:
Is the sleep necessary? q.get should block until the message arrives.

Reply:
fixed